IntegrationTests/PurchaseTester/PurchaseTester/CatsViewController.swift

```diff
@@ -63,6 +68,46 @@ class CatsViewController: UIViewController {
     @objc func goPremiumButtonTapped() {
         navigationController?.popViewController(animated: true)
     }
+
+    @objc func manageSubButtonTapped() {
+        Purchases.shared.showManageSubscriptionModal { maybeError in
+            if let error = maybeError {
+                print("Error opening Manage Sub Modal: \(error.localizedDescription)")
+            }
+        }
+    }
+
+    @objc func beginRefundButtonTapped() {
+        if #available(iOS 15.0, *) {
+            // refund for pro_cat's productID
+            _ = Task<Void, Never> {
+                do {
+                    let customerInfo = try await Purchases.shared.customerInfo()
+                    guard let activeEntitlement = customerInfo.entitlements.active.first else {
+                        print("no current entitlement available, can't begin a refund request!")
+                        return
+                    }
+
+                    let productID = activeEntitlement.value.productIdentifier
+
+                    Purchases.shared.beginRefundRequest(for: productID)
+                    { status, maybeError in
+                        switch status {
+                        case .success: print("Refund request submitted!")
+                        case .userCancelled: print("Refund request cancelled")
+                        case .error: print("Issue submitting refund request: \(maybeError?.localizedDescription ?? "")")
+                        }
+                    }
+
+                }
+                catch {
+                    print("Couldn't fetch customerInfo! Details: \(error.localizedDescription)")
```

Comment:
@beylmk @NachoSoto I just added this to replace the hardcoded entitlement ID

Reply:
could we just save the customerInfo that we get in `viewDidLoad`?

Reply:
either way thanks for fixing this 👍